straxen/plugins/acqmon_processing.py

```diff
@@ -56,48 +133,50 @@ def compute(self, raw_records_aqmon):
-    """ Find pairs of veto start and veto stop signals and the veto duration between them
-    busy_*  <= V1495 busy veto for tpc channels
-    he_*    <= V1495 busy veto for high energy tpc channels
-    hev_*   <= DDC10 hardware high energy veto
+    """ Find pairs of veto start and veto stop signals and the veto
+    duration between them:
+        busy_*  <= V1495 busy veto for tpc channels
+        he_*    <= V1495 busy veto for high energy tpc channels
+        hev_*   <= DDC10 hardware high energy veto
+        straxen_deadtime <= special case of deadtime introduced by the
+            DAQReader-plugin
     """
-
-    __version__ = '0.1.6'
-    depends_on = ('aqmon_hits')
-    provides = ('veto_intervals')
-    data_kind = ('veto_intervals')
+    __version__ = '1.0.0'
+    depends_on = 'aqmon_hits'
+    provides = 'veto_intervals'
+    data_kind = 'veto_intervals'
+
+    max_veto_window = straxen.URLConfig(
+        default=int(10e9),
+        track=True,
+        type=int,
+        help='Maximum separation between veto stop and start pulses [ns]'
+    )
 
     def infer_dtype(self):
         dtype = [(('veto interval [ns]', 'veto_interval'), np.int64),
-                 (('veto signal type', 'veto_type'), np.str_('U9'))]
+                 (('veto signal type', 'veto_type'), np.str_('U20'))]
         dtype += strax.time_fields
         return dtype
 
     def setup(self):
         self.veto_names = ['busy_', 'he_', 'hev_']
-        self.channel_map = {name: ch + straxen.n_hard_aqmon_start for ch, name in
-                            enumerate(['sum_wf', 'm_veto_sync',
-                                       'hev_stop', 'hev_start',
-                                       'he_stop', 'he_start',
-                                       'busy_stop', 'busy_start'])}
+        self.channel_map = {aq_ch.name.lower(): int(aq_ch)
+                            for aq_ch in AqmonChannels}
 
     def get_window_size(self):
         # Give a very wide window
-        return int(self.config['max_veto_window'] * 100)
+        return int(self.config['max_veto_window'])
 
     def compute(self, aqmon_hits, start, end):
-        hits = aqmon_hits
+        # Allocate a nice big buffer and throw away the part we don't need later
         result = np.zeros(len(aqmon_hits) * len(self.veto_names), self.dtype)
         vetos_seen = 0
 
-        for name in self.veto_names:
-            veto_hits_start = channel_select_(hits, self.channel_map[name + 'start'])
-            veto_hits_stop = channel_select_(hits, self.channel_map[name + 'stop'])
+        for veto_name in self.veto_names:
+            veto_hits_start = channel_select_(aqmon_hits, self.channel_map[veto_name + 'start'])
+            veto_hits_stop = channel_select_(aqmon_hits, self.channel_map[veto_name + 'stop'])
 
             # Here we rely on the fact that for each start, there is a single stop that
             # follows it in time. If this is not true, our hardware does not work.
```

Comment:
I do not get the part which follows here. If there should only for each start a single stop why not just subtracting them from each other as they are?

Reply:
yeah, let me look into this indeed, there were edge cases due to the bug I fixed above, but there are also edge cases here at the start/stop of a run as @RoBGlaBe found where we miss one start or one stop, but let me add that logic here